File: TypeScript/src/accounts/student.ts
import { PaymentAccount } from "./paymentaccount";

/**
 * Refactor Exercise 2: Account
 *
 * @author Zishen Wen (F22), Deyuan Chen (S22)
 */
// TODO: Think about what can be a issue of the Student class?
class Student{

	private payment : PaymentAccount;
	private gpa: number;

	constructor(payment: PaymentAccount, gpa: number) {
		this.gpa = gpa;
		this.payment = payment;
	}

	public getGPA(): number {
		return this.gpa;
	}

	public updateGPA(newGPA: number) {
		this.gpa = newGPA;
	}

	// TODO: Do you think student should have an method of pay? (Refused Bequest)
	public pay(amount: number): boolean {
		return this.payment.pay(amount);
	}
}

File: Java/src/main/java/account/Student.java
package account;

/**
 * Refactor Exercise 2: Account
 *
 * @author Zishen Wen (F22), Deyuan Chen (S22)
 */
public class Student {

    private final PaymentAccount paymentAccount;
    private double gpa;

    public Student(PaymentAccount accountData, double gpa) {
        this.paymentAccount = accountData;
        this.gpa = gpa;
    }

    public void updateGPA(double gpa) {
        this.gpa = gpa;
    }

    public double getGPA() {
        return this.gpa;
    }

    public boolean pay(int amount) {
        return paymentAccount.pay(amount);
    }
}
